istioctl/cmd/workload.go

```diff
@@ -0,0 +1,402 @@
+	configureCmd.PersistentFlags().StringVarP(&namespace, "namespace", "n", "", "The namespace that the workload instances belongs to")
+	configureCmd.PersistentFlags().StringVarP(&outputDir, "output", "o", "", "Output directory for generated files")
+	configureCmd.PersistentFlags().StringVar(&clusterID, "clusterID", "", "The ID used to identify the cluster")
+	configureCmd.PersistentFlags().Int64Var(&tokenDuration, "tokenDuration", 3600, "The token duration in seconds (default: 1 hour)")
+	configureCmd.PersistentFlags().StringVar(&revision, "revision", "", "control plane revision (experimental)")
+	return configureCmd
+}
+
+func readWorkloadGroup(filename string, wg *clientv1alpha3.WorkloadGroup) error {
+	f, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+	if err = yaml.Unmarshal(f, wg); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Creates all the relevant config for the given workload group and cluster
+func createConfig(kubeClient kube.ExtendedClient, wg *clientv1alpha3.WorkloadGroup, clusterID, revision, outputDir string) error {
+	if err := validateWorkloadGroup(wg); err != nil {
+		return err
+	}
+	if err := os.MkdirAll(outputDir, filePerms); err != nil {
+		return err
+	}
+	if err := createClusterEnv(wg, outputDir); err != nil {
+		return err
+	}
+	if err := createCertsTokens(kubeClient, wg, outputDir); err != nil {
+		return err
+	}
+	meshConfig, err := createMeshConfig(kubeClient, wg, clusterID, revision, outputDir)
+	if err != nil {
+		return err
+	}
+	if err := createHosts(kubeClient, wg, meshConfig, outputDir); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Used to populate nil WorkloadGroup structs (metadata and template)
+// Additionally checks the invalid fields in template
+func validateWorkloadGroup(wg *clientv1alpha3.WorkloadGroup) error {
+	if wg.Spec.Metadata == nil {
+		wg.Spec.Metadata = &networkingv1alpha3.WorkloadGroup_ObjectMeta{}
+	}
+	if wg.Spec.Template == nil {
+		wg.Spec.Template = &networkingv1alpha3.WorkloadEntry{}
+	}
+	// error checks and warnings
+	template := wg.Spec.Template
+	if template.Address != "" {
+		return fmt.Errorf("address %s should not be set in the WorkloadEntry template", template.Address)
+	}
+	if len(template.Labels) != 0 {
+		fmt.Printf("Labels should be set in the metadata. The following WorkloadEntry labels will override metadata labels: %s", template.Labels)
+	}
+	// default service account for an empty field is "default"
+	if template.ServiceAccount == "" {
+		template.ServiceAccount = "default"
+	}
+	return nil
+}
+
+// Write cluster.env into the given directory
+func createClusterEnv(wg *clientv1alpha3.WorkloadGroup, dir string) error {
+	we := wg.Spec.Template
+	ports := []string{}
+	for _, v := range we.Ports {
+		ports = append(ports, fmt.Sprint(v))
+	}
+	// respect the inbound port annotation and capture all traffic if no inbound ports are set
+	portBehavior := "*"
```

Comment:
@rshriram @costinm I feel like we should have two modes:
* Capture all ports (blackbox)
* Capture no ports (whitebox)

Having this mixed mode where we capture some ports but not others is what we used to do and migrated off of for k8s. It adds a weird coupling on containerPort (here, workloadEntry.ports) when there should not be done. WDYT?

Basically my suggestion is to always use `*`. If they want whitebox mode they can add the annotation to disable redirect on the workload group (that part can definitely be done in another pr)